pkg/azuredisk/azure_managedDiskController.go

```diff
@@ -414,6 +414,79 @@ func (c *ManagedDiskController) ResizeDisk(ctx context.Context, diskURI string,
 	return newSizeQuant, nil
 }
 
+// ModifyDisk: modify disk
+func (c *ManagedDiskController) ModifyDisk(ctx context.Context, options *ManagedDiskOptions) error {
+	var err error
+	klog.V(4).Infof("azureDisk - modifying managed Name:%s, StorageAccountType:%s, DiskIOPSReadWrite:%s, DiskMBpsReadWrite:%s", options.DiskName, options.StorageAccountType, options.DiskIOPSReadWrite, options.DiskMBpsReadWrite)
+
+	rg, subsID, err := getInfoFromDiskURI(options.SourceResourceID)
+	if err != nil {
+		return err
+	}
+
+	diskClient, err := c.clientFactory.GetDiskClientForSub(subsID)
+	if err != nil {
+		return err
+	}
+
+	result, err := diskClient.Get(ctx, rg, options.DiskName)
+	if err != nil {
+		return err
+	}
+
+	if result.Properties == nil || result.SKU == nil || result.SKU.Name == nil {
+		return fmt.Errorf("DiskProperties or SKU of disk(%s) is nil", options.DiskName)
+	}
+
+	diskSku := *result.SKU.Name
+	if options.StorageAccountType != "" {
+		diskSku = options.StorageAccountType
+	}
+
+	diskProperties := armcompute.DiskUpdateProperties{}
+
+	if diskSku == armcompute.DiskStorageAccountTypesUltraSSDLRS || diskSku == armcompute.DiskStorageAccountTypesPremiumV2LRS {
+		if options.DiskIOPSReadWrite != "" {
+			v, err := strconv.Atoi(options.DiskIOPSReadWrite)
+			if err != nil {
+				return fmt.Errorf("AzureDisk - failed to parse DiskIOPSReadWrite: %w", err)
+			}
+			diskIOPSReadWrite := int64(v)
+			diskProperties.DiskIOPSReadWrite = pointer.Int64(diskIOPSReadWrite)
+		}
+
+		if options.DiskMBpsReadWrite != "" {
+			v, err := strconv.Atoi(options.DiskMBpsReadWrite)
+			if err != nil {
+				return fmt.Errorf("AzureDisk - failed to parse DiskMBpsReadWrite: %w", err)
+			}
+			diskMBpsReadWrite := int64(v)
+			diskProperties.DiskMBpsReadWrite = pointer.Int64(diskMBpsReadWrite)
+		}
+	} else {
+		if options.DiskIOPSReadWrite != "" {
+			return fmt.Errorf("AzureDisk - DiskIOPSReadWrite parameter is only applicable in UltraSSD_LRS disk type")
+		}
+		if options.DiskMBpsReadWrite != "" {
+			return fmt.Errorf("AzureDisk - DiskMBpsReadWrite parameter is only applicable in UltraSSD_LRS disk type")
+		}
+	}
+
+	model := armcompute.DiskUpdate{
+		SKU: &armcompute.DiskSKU{
```

Comment:
in L432, there is `result, err := diskClient.Get(ctx, rg, options.DiskName)` operation, is this necessary if disk sku is not going to be modified?

Reply:
I move `		model.SKU = &armcompute.DiskSKU{
			Name: to.Ptr(diskSku),
		}` and `model.Properties = &diskProperties` in update needed. But I think `result, err := diskClient.Get(ctx, rg, options.DiskName)` is still need because we need old disk sku in L443 and L454